org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurSyntaxWidget.java

```diff
@@ -759,6 +768,11 @@ boolean saveAs(String filename, Collection<String> bannedNames) {
         }
         this.filename = Util.canon(filename); // a new file's canonical name may
                                              // change
+        try{
+            fileModifiedDate = Util.getModifiedDate(this.filename);
```

Comment:
when can this throw?  You seem to catch Exception only here and nowhere else, so I don't understand what's so special about this case here.

Reply:
You are correct, File.lastModified() is not supposed to throw, it instead returns 0 on failure. I can change the code to this:
```java
fileModifiedDate = Util.getModifiedDate(this.filename);
if (fileModifiedDate == 0)
    fileModifiedDate = new Date().getTime();
```
How does it look?